compiler/src/test/kotlin/utility.kt

```diff
@@ -0,0 +1,104 @@
+package dk.aau.cs.d409f19
+
+import dk.aau.cs.d409f19.cellumata.CompilerData
+import dk.aau.cs.d409f19.cellumata.ast.*
+import dk.aau.cs.d409f19.cellumata.visitors.LiteralExtractorVisitor
+import dk.aau.cs.d409f19.cellumata.visitors.ScopeCheckVisitor
+import dk.aau.cs.d409f19.cellumata.visitors.TypeChecker
+import org.antlr.v4.runtime.CharStreams
+import org.antlr.v4.runtime.CommonTokenStream
+
+class Utilities {
+
+    companion object {
+
+        /**
+         * Compiles a program given as string parameter
+         */
+        fun compileProgram(program: String): CompilerData {
+            val source = CharStreams.fromString(program)
+            val lexer = dk.aau.cs.d409f19.antlr.CellmataLexer(source)
+            val tokenStream = CommonTokenStream(lexer)
+            val parser = dk.aau.cs.d409f19.antlr.CellmataParser(tokenStream)
+
+            // Build AST
+            val startContext = parser.start()
+            val ast = reduce(startContext)
+
+            // Extract literals
+            LiteralExtractorVisitor().visit(ast)
+
+            // Symbol table and scope
+            val scopeChecker = ScopeCheckVisitor()
+            scopeChecker.visit(ast)
+            val symbolTable = scopeChecker.getSymbolTable()
+
+            // Type checking
+            TypeChecker(symbolTable).visit(ast)
+
+            return CompilerData(parser, ast, scopeChecker)
+        }
+
+        fun getParser(program: String): dk.aau.cs.d409f19.antlr.CellmataParser {
+            val source = CharStreams.fromString(program)
+            val lexer = dk.aau.cs.d409f19.antlr.CellmataLexer(source)
+            val tokenStream = CommonTokenStream(lexer)
+            return dk.aau.cs.d409f19.antlr.CellmataParser(tokenStream)
+        }
+
+        /**
+         * Returns a world declaration based on given parameters and with sane defaults
+         * TODO: should handle optional, second dimension - as of now, it's required
+         */
+        fun getWorldDecl(
+            dimOneSize: Int = 10,
+            dimOneType: String = "wrap",
+            dimTwoSize: Int = 20,
+            dimTwoType: String = "wrap",
+            tickrate: Int = 120,
+            cellsize: Int = 5
+        ): String {
+            return """world {
+            |  size = $dimOneSize[$dimOneType], $dimTwoSize[$dimTwoType]
```

Comment:
What about single dimensional worlds?

Reply:
Yes, exactly. I'm not sure how to do that in Kotlin. In the javadoc to exactly this function is says:
`* TODO: should handle optional, second dimension - as of now, it's required`.

Any good ideas?

Reply:
Either make `dimTwoSize` and `dimTwoType` optional, or use a list for dimensions

Reply:
Fixed on 2818d968cee8a03f28d3f0ebce7a848c5c6e8958

Reply:
Fixed in 540ed119135e0e4be25091787731dbbb365eb1a3

Reply:
Why didn't you use optionals or lists?

Reply:
Because this worked with the least code